### PR TITLE
Refactor backend test suite of Reduction ops into template plugin tests

### PR DESCRIPTION
**Pull request by ggalieroc**
### Details:
 - *Refactor visitors test to print operation within test name*
 - *Refactor backend test suite of Reduction ops into template plugin tests*

### Tickets:
 - *N/A*

### CODE REVIEW

File: docs/template_plugin/tests/functional/op_reference/reduce_l2.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+namespace {
+
+std::vector<ReductionParams> reductionParams = {
+    ReductionParams(PartialShape{3, 2, 2}, element::f32,
+                    std::vector<float>{1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0, 10.0, 11.0, 12.0},
+                    std::vector<int64_t>{2},
+                    std::vector<float>{2.23606798, 5.0, 7.81024968, 10.63014581, 13.45362405, 16.2788206},
+                    true,
+                    ReductionType::L2),
+    ReductionParams(PartialShape{3, 2, 2}, element::f32,
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/reduce_l2.cpp:29]: Could we cover all supported data types?

File: docs/template_plugin/tests/functional/op_reference/reduce_mean.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+std::vector<ReductionParams> generateReductionCombinedParams() {
+    const std::vector<std::vector<ReductionParams>> reductionTypeParams {
+        generateReductionParams<element::Type_t::f32>(true),
+        generateReductionParams<element::Type_t::f32>(false),
+        generateReductionParams<element::Type_t::f16>(true),
+        generateReductionParams<element::Type_t::f16>(false),
+        generateReductionParams<element::Type_t::i32>(true),
+        generateReductionParams<element::Type_t::i32>(false),
+        generateReductionParams<element::Type_t::u32>(true),
+        generateReductionParams<element::Type_t::u32>(false)
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/reduce_mean.cpp:59]: u64?

**Review by ilyachur — COMMENTED** [review]
As I see some data types are not covered. Could we extend tests?

File: docs/template_plugin/tests/functional/op_reference/reduce_l2.cpp (hunk 1 above)
Comment (ggalieroc) [reply docs/template_plugin/tests/functional/op_reference/reduce_l2.cpp]: Done

File: docs/template_plugin/tests/functional/op_reference/reduce_mean.cpp (hunk 2 above)
Comment (ggalieroc) [reply docs/template_plugin/tests/functional/op_reference/reduce_mean.cpp]: Extended

File: docs/template_plugin/tests/functional/op_reference/reduction.hpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+struct ReductionParams {
+    template <class ET>
+    ReductionParams(const ngraph::PartialShape& inShape, const ngraph::element::Type& inType, const std::vector<ET> inValues,
+                    const std::vector<int64_t> axesValues, const std::vector<ET> outValues, const bool keepDims,
+                    const ngraph::helpers::ReductionType& reductType)
+            : dataShape(inShape), elemType(inType), inputData(CreateBlob(inType, inValues)), axesData(axesValues), refData(CreateBlob(inType, outValues)),
+              keepDimensions(keepDims), reductionType(reductType) {}
+    ngraph::PartialShape dataShape;
+    ngraph::element::Type elemType;
+    InferenceEngine::Blob::Ptr inputData;
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/reduction.hpp:28]: We have a Tensor struct which joins these structures.

**Review by ilyachur — COMMENTED** [review]
Could you fix CI?

File: docs/template_plugin/tests/functional/op_reference/reduction.hpp (hunk 3 above)
Comment (ggalieroc) [reply docs/template_plugin/tests/functional/op_reference/reduction.hpp]: Good point. I have changed it to use Tensor for data input and output. As for axes I left vector as I want it to be part of the test case name

File: ngraph/test/runtime/interpreter/unit_test.manifest [hunk 4]
```diff
@@ -30,16 +26,6 @@ INTERPRETER.onnx_model_matmul_integer_4d_no_zero_point
...
-INTERPRETER.reduce_sum_2d_to_scalar_int8
-INTERPRETER.reduce_sum_keep_2d_to_scalar_int8
-INTERPRETER.reduce_product_to_scalar_int8
-INTERPRETER.reduce_product_keep_to_scalar_int8
-INTERPRETER.reduce_min_to_scalar_int8
-INTERPRETER.reduce_min_keep_to_scalar_int8
-INTERPRETER.reduce_mean_to_scalar_int8
-INTERPRETER.reduce_mean_keep_to_scalar_int8
-INTERPRETER.reduce_max_to_scalar_int8
-INTERPRETER.reduce_max_keep_to_scalar_int8
```
Comment (mitruska) [inline ngraph/test/runtime/interpreter/unit_test.manifest:42]: Do we have a replacement for these tests?

File: ngraph/test/runtime/ie/unit_test.manifest [hunk 5]
```diff
@@ -357,28 +357,6 @@ all_trivial
...
-reduce_min_keep_to_scalar_int8
-# accuracy
-reduce_mean_matrix_rows_int32
-reduce_mean_keep_matrix_rows_int32
-reduce_sum_keep_stable_acc
-reduce_sum_keep_3d_to_scalar_int32
-reduce_sum_keep_large_1d_to_scalar
-reduce_sum_stable_acc
-reduce_sum_3d_to_scalar_int32
-reduce_sum_large_1d_to_scalar
```
Comment (mitruska) [inline ngraph/test/runtime/ie/unit_test.manifest:380]: As it was disabled and reported for CPU, do we have such Single Layer Tests?

**Review by ilyachur — APPROVED** [review]

**Pull request merged**